src/main/java/com/google/cloud/tools/app/config/impl/DefaultStageGenericJavaConfigurationImpl.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.app.config.impl;
+
+import com.google.cloud.tools.app.config.StageGenericJavaConfiguration;
+
+import java.nio.file.Path;
+
+import javax.annotation.Nullable;
+
+/**
+ * Default implementation for
+ * {@link com.google.cloud.tools.app.config.StageGenericJavaConfiguration}.
+ */
+public class DefaultStageGenericJavaConfigurationImpl implements StageGenericJavaConfiguration {
+
+  private final Path appYaml;
+  private final Path dockerfile;
+  private final Path artifact;
+  private final Path stagingDirectory;
+
+  public DefaultStageGenericJavaConfigurationImpl(@Nullable Path appYaml, @Nullable Path dockerfile,
+      Path artifact, Path stagingDirectory) {
+    this.appYaml = appYaml;
+    this.dockerfile = dockerfile;
+    this.artifact = artifact;
+    this.stagingDirectory = stagingDirectory;
+  }
+
+  public Path getAppYaml() {
+    return appYaml;
+  }
+
+  public Path getDockerfile() {
+    return dockerfile;
+  }
+
+  public Path getArtifact() {
+    return artifact;
+  }
+
+  public Path getStagingDirectory() {
+    return stagingDirectory;
+  }
+
+  public static class Builder {
+    private Path appYaml;
```

Comment:
@patflynn @loosebazooka what are you guys' thoughts on repeated members in builder and configuration object, so the config members are all final? Do you have any suggestion on how to make this look better?